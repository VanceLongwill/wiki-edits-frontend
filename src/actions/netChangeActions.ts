import { LangCode } from '../types'

export const FETCH_NET_CHANGE = 'FETCH_NET_CHANGE'
export const FETCH_NET_CHANGE_SUCCESS = 'FETCH_NET_CHANGE_SUCCESS'
export const FETCH_NET_CHANGE_FAIL = 'FETCH_NET_CHANGE_FAIL'
export const PERIODICALLY_FETCH_NET_CHANGE = 'PERIODICALLY_FETCH_NET_CHANGE'

export interface IFetchNetChange {
  type: typeof FETCH_NET_CHANGE
  payload: {
    from: number
    to: number
    langCode: LangCode
  }
}

export interface IFetchNetChangeFail {
  type: typeof FETCH_NET_CHANGE_FAIL
  payload: {
    from: number
    to: number
    langCode: LangCode
    error: string
  }
}

export interface IFetchNetChangeSuccess {
  type: typeof FETCH_NET_CHANGE_SUCCESS
  payload: {
    from: number
    to: number
    langCode: LangCode
    changeInBytes: number
  }
}

export function fetchNetChange(
  langCode: LangCode,
  from: number,
  to: number
): IFetchNetChange {
  return {
    payload: {
      langCode,
      from,
      to,
    },
    type: FETCH_NET_CHANGE,
  }
}

export function fetchNetChangeFail(
  langCode: LangCode,
  from: number,
  to: number,
  error: string
): IFetchNetChangeFail {
  return {
    payload: { langCode, from, to, error },
    type: FETCH_NET_CHANGE_FAIL,
  }
}

export function fetchNetChangeSuccess(
  langCode: LangCode,
  from: number,
  to: number,
  changeInBytes: number
): IFetchNetChangeSuccess {
  return {
    payload: { langCode, to, from, changeInBytes },
    type: FETCH_NET_CHANGE_SUCCESS,
  }
}

export type NetChangeActions =
  | IFetchNetChangeFail
  | IFetchNetChangeSuccess
  | IFetchNetChange
