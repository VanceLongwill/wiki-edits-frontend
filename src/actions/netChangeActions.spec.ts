import * as actions from './netChangeActions'

describe('netChangeActions', () => {
  it('action creators should match snapshot', () => {
    expect(actions.fetchNetChange('en', 123233222, 123123123)).toMatchSnapshot()
    expect(
      actions.fetchNetChangeSuccess('en', 121113222, 111123123, 123123)
    ).toMatchSnapshot()
    expect(
      actions.fetchNetChangeFail('en', 123233782, 625123123, 'some error')
    ).toMatchSnapshot()
  })
})
