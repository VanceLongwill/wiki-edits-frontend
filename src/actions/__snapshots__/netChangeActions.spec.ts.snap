// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`netChangeActions action creators should match snapshot 1`] = `
Object {
  "payload": Object {
    "from": 123233222,
    "langCode": "en",
    "to": 123123123,
  },
  "type": "FETCH_NET_CHANGE",
}
`;

exports[`netChangeActions action creators should match snapshot 2`] = `
Object {
  "payload": Object {
    "changeInBytes": 123123,
    "from": 121113222,
    "langCode": "en",
    "to": 111123123,
  },
  "type": "FETCH_NET_CHANGE_SUCCESS",
}
`;

exports[`netChangeActions action creators should match snapshot 3`] = `
Object {
  "payload": Object {
    "error": "some error",
    "from": 123233782,
    "langCode": "en",
    "to": 625123123,
  },
  "type": "FETCH_NET_CHANGE_FAIL",
}
`;
